.pre-commit-config.yaml

```diff
@@ -5,17 +5,22 @@ repos:
     rev: v4.0.1
     hooks:
       - id: check-merge-conflict
+      - id: check-added-large-files
+      - id: check-case-conflict
       - id: check-json
       - id: check-toml
+      - id: check-xml
       - id: check-yaml
         exclude: 'mkdocs.yml' # Exclude all mkdocs.yml as they use tags i.e. `!!!`
       - id: pretty-format-json
-        args: ['--indent=4','--autofix']
+        args: [--indent=4, --autofix]
       - id: end-of-file-fixer
       - id: no-commit-to-branch
+        name: protect-branch
```

Comment:
This is just an added alias to the current protection. We have had the local protection for a while now.
And repo Admins are still allowed to push to main iirc. This just means that no one will commit to main locally and ensure everyone is already on a feature branch.

Reply:
Nope, if you have checked to allow protection for admins (or a similar checkmark) then this would also work for admins.

I mean it isn't our task to make sure contributors commit to scope branches only, and git won't allow them to push anyways so they will change their branch and remember it the next time. I would like to keep pre-commit a short task.

Reply:
I will admit (reluctantly) this has saved me multiple times from commiting to main locally and having to sort it out on to a new branch, which while its not hard to do, is just a few more commands than this saves. IMO this is worth having, and we have had it for a while. 

I'm more inclined to not add something provided in this pull request than I am to remove a current one.

Reply:
Again, like said above I am not sure if it would be a nice addition, and it is two commands, `git branch -b feature/feature; git checkout main; git reset --hard HEAD`.

Reply:
Please reread what I've said. This is not an addition, `no-commit-to-branch` been part of the configuration for a while.

Reply:
Yes I know, I am talking about it in general.